packages/zipkin/src/batch-recorder.js

```diff
@@ -6,8 +6,14 @@ function PartialSpan(traceId) {
   this.startTimestamp = now();
   this.startTick = hrtime();
   this.delegate = new Span(traceId);
-  this.localEndpoint = new Endpoint({serviceName: 'unknown'});
+  this.localEndpoint = new Endpoint({});
```

Comment:
the batch recorder shouldn't be setting this.. it should always be explicitly set by either instrumentation or the tracer (otherwise there's a difference in behavior when reading the console logger vs the batch recorder)